samza-azure/src/main/java/org/apache/samza/coordinator/AzureCoordinationServiceFactory.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.coordinator;
+
+import org.apache.samza.config.Config;
+
+public class AzureCoordinationServiceFactory implements CoordinationServiceFactory {
+
+  @Override
+  public CoordinationUtils getCoordinationService(String groupId, String participantId, Config config) {
+    return new AzureCoordinationUtils(config);
```

Comment:
The lock name is currently the lock path which is the hash of the ExecutionPlan. This is passed as a string to the getLock method in AzureCoordinationUtils. Apart from this, I need the container name and the the blob name which are generated using the globalAppId in AzureConfig. Let me know if this is not the right way to do it and I'm misinterpreting something here.